patroni/postgresql/config.py

```diff
@@ -487,8 +497,7 @@ def write_postgresql_conf(self, configuration: Optional[CaseInsensitiveDict] = N
             include = self._config.get('custom_conf') or self._postgresql_base_conf_name
             f.writeline("include '{0}'\n".format(ConfigWriter.escape(include)))
             for name, value in sorted((configuration).items()):
-                value = transform_postgresql_parameter_value(self._postgresql.major_version, name, value,
-                                                             self._postgresql.available_gucs)
+                value = transform_postgresql_parameter_value(self.full_pg_version, name, value)
```

Comment:
Lets introduce a new variable outside the loop to avoid calling `postgres --version` for every parameter in the worst case.